### PR TITLE
Add shebang parsing to cnex

### PR DESCRIPTION
**Pull request by gitlarryf**

### CODE REVIEW

File: exec/cnex/bytecode.c [hunk 1]
```diff
@@ -109,7 +109,11 @@ void bytecode_loadBytecode(TBytecode *b, const uint8_t *bytecode, unsigned int l
 
     assert(b != NULL);
 
-    if (memcmp(bytecode, "Ne\0n", 4) != 0) {
+    if (memcmp(bytecode, "#!", 2) == 0) {
+        while (bytecode[i] != '\n' && i < len) i++;
```
Comment (ghewgill) [inline exec/cnex/bytecode.c:113]: This needs to compare `i < len` *before* accessing the array. Also, use curly braces even though you've only got an increment statement.

**Review by ghewgill — CHANGES_REQUESTED** [review]
This needs test support.

File: exec/cnex/bytecode.c [hunk 2]
```diff
@@ -109,7 +109,16 @@ void bytecode_loadBytecode(TBytecode *b, const uint8_t *bytecode, unsigned int l
 
     assert(b != NULL);
 
-    if (memcmp(bytecode, "Ne\0n", 4) != 0) {
+    if (memcmp(bytecode, "#!", 2) == 0) {
+        while (i < len && bytecode[i] != '\n') {
+            i++;
+        }
+        if (i+1 >= len) {
```
Comment (ghewgill) [inline exec/cnex/bytecode.c:116]: Instead of using `i+1` here and incrementing after, it would be cleaner to increment first, then check `i >= len`.

**Review by ghewgill — CHANGES_REQUESTED** [review]
This still needs a test to actually test this functionality.

File: exec/cnex/bytecode.c (hunk 2 above)
Comment (gitlarryf) [reply exec/cnex/bytecode.c]: Fixed.

File: exec/cnex/bytecode.c (hunk 1 above)
Comment (gitlarryf) [reply exec/cnex/bytecode.c]: Fixed.

**Pull request closed without merging**